### DISCUSSION TITLE
fzf showed up twice when I assign the output to a variable

### DISCUSSION THREAD

**Discussion by FranklinYu** [junegunn/fzf#3555] [Q&A] [answered]
When I do

```sh
selection=$(find * -type f | fzf)
```

I saw fzf showing up twice; `$selection` was assigned the value of the first selection. Did anyone come across this issue?

**Comment by LangLangBart** [comment, marked as answer]
Not reproducible. You could share the fzf version, system settings (shell/OS), and check if you have assigned `fzf` differently.

```bash
type -a fzf
# fzf is an alias for fzf  --bind "enter:reload:find * -type f"
# fzf is /usr/local/bin/fzf
# fzf is /Users/paria/.cache/zsh4humans/v5/fzf/bin/fzf
```

Also, share your `FZF_DEFAULT_COMMAND` and `FZF_DEFAULT_OPTS`.


```bash
printenv FZF_DEFAULT_OPTS
# …
```

any plugins ?

**Reply by FranklinYu** [reply]
Thanks, indeed it is part of my Zsh configuration. I have not fully understood the problem, but somehow this fixed it:

https://github.com/FranklinYu/chezmoi-data/commit/4388fc8c3452832663f04a169b5150acb4935c52

My current theory is that Zsh’s understanding of the arithmetic expression is different from what I intend. It is still confusing why this bug wasn’t triggered when I do

```sh
find * -type f | fzf | read selection
```

instead.

(Marking this as the answer to “close” this question. If I eventually figure out exactly why, I may post it here.)
